data-prepper-plugins/obfuscate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/obfuscation/action/OneWayHashActionConfig.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.obfuscation.action;
+
+import java.nio.charset.StandardCharsets;
+import java.security.SecureRandom;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyDescription;
+
+import jakarta.validation.constraints.Pattern;
+
+
+public class OneWayHashActionConfig {
+
+
+    @JsonProperty("salt")
+    @JsonPropertyDescription("Salt value to use when generating hash. If not specified, salt will be randomly generated by the processor.")
+    private byte[] salt;
+
+    @JsonProperty("format")
+    @Pattern(regexp = "SHA-512", message = "Valid values: SHA-512")
+    @JsonPropertyDescription("Format of one way hash to generate. Default to SHA-512.")
+    private String hashFormat = "SHA-512";
+
+    @JsonProperty("salt_key")
+    @JsonPropertyDescription("A key to compute salt based on a value provided as part of a record." + 
+        "If key or value was not found in the record(s), a salt defined in the pipeline configuration will be used instead.")
+    private String saltKey;
```

Comment:
Please use an `EventKey` instead of a `String` for the type here. And then use that wherever it is used. This allows us to verify the input early and improves performance.

Feel free to review #4628 as a reference to this approach.

Reply:
Done